domain/src/test/java/uk/gov/hmcts/ccd/definition/store/domain/AppConfigBasedAmPersistenceSwitchTest.java

```diff
@@ -0,0 +1,150 @@
+        doReturn(ccdOnlyReadCaseTypes).when(duplicateConfigForWriteApplicationParams).getCaseTypesWithAmReadFromCcd();
+        doReturn(amOnlyReadCaseTypes).when(duplicateConfigForWriteApplicationParams).getCaseTypesWithAmReadFromAm();
+
+        doReturn(repeatedCaseTypeList).when(duplicateConfigForReadAndWriteApplicationParams)
+                .getCaseTypesWithAmWrittenOnlyToCcd();
+        doReturn(repeatedCaseTypeList).when(duplicateConfigForReadAndWriteApplicationParams)
+                .getCaseTypesWithAmWrittenOnlyToAm();
+        doReturn(repeatedCaseTypeList).when(duplicateConfigForReadAndWriteApplicationParams)
+                .getCaseTypesWithAmWrittenToBoth();
+        doReturn(repeatedCaseTypeList).when(duplicateConfigForReadAndWriteApplicationParams)
+                .getCaseTypesWithAmReadFromCcd();
+        doReturn(repeatedCaseTypeList).when(duplicateConfigForReadAndWriteApplicationParams)
+                .getCaseTypesWithAmReadFromAm();
+    }
+
+    @Test
+    public void shouldSelectCcdAsReadDataSourceIfCaseTypeNotSpecified() {
+        assertEquals(FROM_CCD, goodAmPersistenceSwitch.getReadDataSourceFor(UNSPECIFIED));
+    }
+
+    @Test
+    public void shouldSelectCcdAsWriteDataSourceIfCaseTypeNotSpecified() {
+        assertEquals(TO_CCD, goodAmPersistenceSwitch.getWriteDataSourceFor(UNSPECIFIED));
+    }
+
+    @Test
+    public void shouldSelectCcdAsReadDataSourceForAppropriateCaseTypes() {
+        assertEquals(FROM_CCD, goodAmPersistenceSwitch.getReadDataSourceFor(CMC_CT));
+        assertEquals(FROM_CCD, goodAmPersistenceSwitch.getReadDataSourceFor(PROBATE_CT));
+    }
+
+    @Test
+    public void shouldSelectAmAsReadDataSourceForAppropriateCaseTypes() {
+        assertEquals(FROM_AM, goodAmPersistenceSwitch.getReadDataSourceFor(DIVORCE_CT));
+        assertEquals(FROM_AM, goodAmPersistenceSwitch.getReadDataSourceFor(FR_CT));
+        assertEquals(FROM_AM, goodAmPersistenceSwitch.getReadDataSourceFor(TEST_CT));
+    }
+
+    @Test
+    public void shouldSelectCcdAsWriteDataSourceForAppropriateCaseTypes() {
+        assertEquals(TO_CCD, goodAmPersistenceSwitch.getWriteDataSourceFor(DIVORCE_CT));
+        assertEquals(TO_CCD, goodAmPersistenceSwitch.getWriteDataSourceFor(CMC_CT));
+        assertEquals(TO_CCD, goodAmPersistenceSwitch.getWriteDataSourceFor(TEST_CT));
+    }
+
+    @Test
+    public void shouldSelectAmAsWriteDataSourceForAppropriateCaseTypes() {
+        assertEquals(TO_AM, goodAmPersistenceSwitch.getWriteDataSourceFor(PROBATE_CT));
+    }
+
+    @Test
+    public void shouldSelectBothAsWriteDataSourceForAppropriateCaseTypes() {
+        assertEquals(TO_BOTH, goodAmPersistenceSwitch.getWriteDataSourceFor(FR_CT));
+    }
+
+    @Test(expected = InvalidPropertyException.class)
+    public void shouldFailForDuplicateReadConfigurations() {
+        new AppConfigBasedAmPersistenceSwitch(
```

Comment:
Could we use `Assertions.assertThrows` from `junit.jupiter` api to go with convention?

Reply:
Also using the assertThrows we could do:
```
final InvalidPropertyException invalidPropertyException = assertThrows(InvalidPropertyException.class, () -> new AppConfigBasedAmPersistenceSwitch(duplicateConfigForReadApplicationParams));
```
and then assert the actual property that is failing as part of `InvalidPropertyException.propertyName` stored value.

Reply:
Done